app/Http/Controllers/Users/CartController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Http\Controllers\Users;
+
+use App\Http\Controllers\Controller;
+use App\Models\Cart;
+use App\Models\Product;
+use Illuminate\Http\Request;
+
+class CartController extends Controller
+{
+    public function index()
+    {
+
+        return view('users.cart.index');
+
+    }
+
+    public function addToCart(Product $product)
+    {
+        if (session()->has('cart'))
+        {
+            $cart = new Cart(session()->get('cart'));
+        } else {
+            $cart = new Cart();
+        }
+        $cart->add($product);
+        session()->put('cart', $cart);
```

Comment:
Theo em thì vì sao khi add Cart như này mình lại dùng Session để lưu mà không tạo luôn 1 bảng Cart trong database?

Reply:
Dạ vì users có thể add to cart nhiều sp( có thể sẽ không mua) điều này làm nặng database.